ui/src/app/machines/components/ActionFormWrapper/CloneForm/CloneFormFields/SourceMachineSelect/SourceMachineSelect.test.tsx

```diff
@@ -0,0 +1,169 @@
+      machine: machineStateFactory({
+        items: [firstMachine, secondMachine],
+        loaded: true,
+        active: null,
+        selected: [],
+      }),
+    });
+  });
+
+  it("dispatches action to fetch machines on load", () => {
+    const store = mockStore(state);
+    mount(
+      <Provider store={store}>
+        <SourceMachineSelect source="" setSource={jest.fn()} />
+      </Provider>
+    );
+
+    expect(
+      store.getActions().some((action) => action.type === "machine/fetch")
+    ).toBe(true);
+  });
+
+  it("shows a spinner while machines are loading", () => {
+    state.machine.loaded = false;
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <SourceMachineSelect source="" setSource={jest.fn()} />
+      </Provider>
+    );
+
+    expect(wrapper.find("[data-test='loading-spinner']").exists()).toBe(true);
+  });
+
+  it("can filter machines by hostname, system_id and/or tags", () => {
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <SourceMachineSelect source="" setSource={jest.fn()} />
+      </Provider>
+    );
+
+    // Filter by "first" which matches the hostname of the first machine
+    wrapper
+      .find("[data-test='source-machine-searchbox'] input")
+      .simulate("change", { target: { value: "first" } });
+    expect(wrapper.find("[data-test='source-machine-first']").exists()).toBe(
+      true
+    );
+    expect(wrapper.find("[data-test='source-machine-second']").exists()).toBe(
+      false
+    );
+
+    // Filter by "def" which matches the system_id of the second machine
+    wrapper
+      .find("[data-test='source-machine-searchbox'] input")
+      .simulate("change", { target: { value: "def" } });
+    expect(wrapper.find("[data-test='source-machine-first']").exists()).toBe(
+      false
+    );
+    expect(wrapper.find("[data-test='source-machine-second']").exists()).toBe(
+      true
+    );
+
+    // Filter by "tag" which matches the tags of the first and second machine
+    wrapper
+      .find("[data-test='source-machine-searchbox'] input")
+      .simulate("change", { target: { value: "tag" } });
+    expect(wrapper.find("[data-test='source-machine-first']").exists()).toBe(
+      true
+    );
+    expect(wrapper.find("[data-test='source-machine-second']").exists()).toBe(
+      true
+    );
+  });
+
+  it("highlights the substring that matches the search text", () => {
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <SourceMachineSelect source="" setSource={jest.fn()} />
+      </Provider>
+    );
+
+    // Filter by "fir" which matches part of the hostname of the first machine
+    wrapper
+      .find("[data-test='source-machine-searchbox'] input")
+      .simulate("change", { target: { value: "fir" } });
+    expect(
+      wrapper.find("[data-test='source-machine-first']")
+    ).toMatchSnapshot();
```

Comment:
Maybe this snapshot masks what you're expecting a bit. Maybe you could do:

```jsx
compareJSX(
      wrapper.find("[data-test='source-machine-first']"),
      <span
        dangerouslySetInnerHTML={{
          __html: "<strong>fir</strong>st",
        }}
        data-test="source-machine-first"
      />
    );
```

Or even

```javascript
expect(
      wrapper
        .find("[data-test='source-machine-first']")
        .render()
        .find("strong")
        .text()
    ).toBe("fir");
```

Reply:
Yeah kinda took the easy option with this test hahaha. I think I'll go with your second recommendation, seems a bit more indicative of what's actually being asserted.